### PR TITLE
Add cluster RecordIO files reader

### LINKED ISSUE #386: Support prepare dataset in fit-a-line demo

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #386

### CODE REVIEW

File: demo/fit_a_line/train.py [hunk 1]
```diff
@@ -2,38 +2,57 @@
...
 import os
 import gzip
+import sys
 
-#PaddleCloud cached the dataset on /pfs/${DATACENTER}/public/dataset/...
-dc = os.getenv("PADDLE_CLOUD_CURRENT_DATACENTER")
-dataset.common.DATA_HOME = "/pfs/%s/public/dataset" % dc
+# NOTE: You should change USERNAME to your account.
+# TODO(Yancey1989): fetch username from environment variable.
+USERNAME = "yanxu05@baidu.com"
```
Comment (typhoonzero) [inline demo/fit_a_line/train.py]: Do not use personal accounts

File: demo/fit_a_line/train.py [hunk 2]
```diff
@@ -2,38 +2,57 @@
...
+#PaddleCloud cached the dataset on /pfs/${DATACENTER}/home/${USERNAME}/...
+dataset.common.DATA_HOME = "/pfs/%s/home/%s" % (DC, USERNAME)
+TRAIN_FILES_PATH = os.path.join(dataset.common.DATA_HOME, "uci_housing")
+
+def prepare_dataset():
+    dataset.common.convert(TRAIN_FILES_PATH,
+                           dataset.uci_housing.train(), 100, "train")
+
+TRAINER_ID = int(os.getenv("PADDLE_INIT_TRAINER_ID"))
+TRAINER_COUNT = int(os.getenv("PADDLE_INIT_NUM_GRADIENT_SERVERS"))
```
Comment (typhoonzero) [inline demo/fit_a_line/train.py]: `TRAINER_COUNT` can be confused with the trainer argument, but it's OK, refine this if you have time.

File: demo/fit_a_line/train.py (hunk 1 above)
Comment (Yancey1989) [reply demo/fit_a_line/train.py]: Done.

File: demo/fit_a_line/train.py (hunk 2 above)
Comment (Yancey1989) [reply demo/fit_a_line/train.py]: Done, maybe `TRAINER_INSTANCES` is more suitable.

**Pull request merged**